includes/Core/Tags/First_Party_Mode/First_Party_Mode.php

```diff
@@ -92,4 +105,36 @@ public function get_debug_fields() {
 			),
 		);
 	}
+
+	/**
+	 * Checks the health of First Party Mode server requirements.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return void
+	 */
+	public function healthcheck() {
```

Comment:
Instead of having the contents of this method duplicated here and in `REST_First_Party_Mode_Controller::get_rest_routes`, as @hussain-t also suggested [here](https://github.com/google/site-kit-wp/issues/9768#issuecomment-2516233777), we should consider relocating the `is_endpoint_healthy` method to this (`First_Party_Mode`) class (as protected method), and re-using this `healthcheck` method within `REST_First_Party_Mode_Controller::get_rest_routes` to avoid code duplication in the spirit of DRY. Please let me know what you think, thanks!